internal/configs/virtualserver.go

```diff
@@ -354,6 +366,23 @@ func generateUpstream(upstreamName string, upstream conf_v1alpha1.Upstream, isEx
 	}
 }
 
+func generateSlowStartForPlus(upstream conf_v1alpha1.Upstream, lbMethod string) string {
```

Comment:
this function needs to be rewritten as it doesn't implement the slow start logic completely. The logic is below:
- if slowStart is "" , return ""
- if the lbMethod is either `random`, `ip_hash` or `hash`, print a warning and return "" . Note that the hash method is defined as hash + some varialble(s), for example `$hash $request_id`. The current implementation incorrectly allows methods like  `$hash $request_id`. Also, the random method is similar to the hash, but has a finite number of possible values. Please see them here -- https://github.com/nginxinc/kubernetes-ingress/blob/master/internal/configs/parsing_helpers.go#L122-L126 Those should be excluded as well.
- return slowStart

Also, the corresponding unit test must be rewritten as well. For the test, consider the following test cases:

Positive cases:
- slow start is "" and the method is "least_conn",  expects  ""
- slow_start is "10s" and the method is "least_conn",  expects "10s"

Negative cases:
- slow_start is "10s" and the method is "hash $something", expects ""
- slow_start is "10s" and the method is "ip_hash", expects ""
- slow_start is "10s" and the method is "random", expects "". Note, this can be repeated for all allowed variations of the random method -- https://github.com/nginxinc/kubernetes-ingress/blob/master/internal/configs/parsing_helpers.go#L122-L126

Please add any additional cases if necessary.

Reply:
Normally we have a preference for using a `map` like: https://github.com/nginxinc/kubernetes-ingress/blob/master/pkg/apis/configuration/validation/validation.go#L343-L357 which I think should be possible.

If sticking with a slice is necessary, you should be able to make it `const` , right?